src/main/java/org/wso2/carbon/module/swiftiso20022/validation/JsonToMT940PayloadValidator.java

```diff
@@ -0,0 +1,291 @@
+    }
+
+    /**
+     * Method to construct list of fields in Balances for length validation.
+     * @param balanceName        Name of the Balance object in Payload
+     * @return List of fields for length validation
+     */
+    public static List<ValidatorContext> getFieldsInBalanceForLengthValidation(String balanceName) {
+        return List.of(
+                new ValidatorContext(MT940Constants.BAL_DATE, StringUtils.join(balanceName,
+                        MT940Constants.DN_DATE), ConnectorConstants.DATE_LENGTH),
+                new ValidatorContext(MT940Constants.BAL_CURRENCY, StringUtils.join(balanceName,
+                        MT940Constants.DN_CURRENCY), ConnectorConstants.CURRENCY_LENGTH),
+                new ValidatorContext(MT940Constants.BAL_AMOUNT, StringUtils.join(balanceName,
+                        MT940Constants.DN_AMOUNT), ConnectorConstants.AMOUNT_LENGTH),
+                new ValidatorContext(MT940Constants.BAL_INDICATOR, StringUtils.join(balanceName,
+                        MT940Constants.DN_INDICATOR), ConnectorConstants.INDICATOR_LENGTH)
+        );
+    }
+
+    // List of alpha fields from the payload for validation.
+    private static List<ValidatorContext> getAlphaParamsInBalances(String balanceName) {
+        return List.of(
+                new ValidatorContext(MT940Constants.BAL_INDICATOR, StringUtils.join(balanceName,
+                        MT940Constants.DN_INDICATOR)),
+                new ValidatorContext(MT940Constants.BAL_CURRENCY, StringUtils.join(balanceName,
+                        MT940Constants.DN_CURRENCY)),
+        new ValidatorContext(MT940Constants.BAL_STATEMENT_TYPE, StringUtils.join(balanceName,
+                MT940Constants.DN_STATEMENT_TYPE))
+        );
+    }
+
+    private static List<ValidatorContext> getNumericParamsInBalances(String balanceName) {
+        return List.of(
+                new ValidatorContext(MT940Constants.BAL_DATE, StringUtils.join(balanceName,
+                        MT940Constants.DN_DATE))
+        );
+    }
+
+    // List of date fields from the payload for validation.
+    private static List<ValidatorContext> getDateParamInBalance(String balanceName) {
+        return List.of(
+                new ValidatorContext(MT940Constants.BAL_DATE, StringUtils.join(balanceName,
+                        MT940Constants.DN_DATE))
+        );
+    }
+
+    // List of currency fields from the payload for validation.
+    private static List<ValidatorContext> getCurrencyParamInBalance(String balanceName) {
+        return List.of(
+                new ValidatorContext(MT940Constants.BAL_CURRENCY, StringUtils.join(balanceName,
+                        MT940Constants.DN_CURRENCY))
+        );
+    }
+
+    /**
+     * Method to construct list of fields in Transactions for Mandatory parameter validation.
+     * @return List of fields for Mandatory parameter validation
+     */
+    public static List<ValidatorContext> getMandatoryFieldsInTransaction() {
```

Comment:
why cant we store them as static constant variables without calling from a method? also check other applicable methods

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/6/commits/02976704c3798e54f6e5b27d539a212b8bdb900d